src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,670 @@
+#define PROTID  0x14;
+#define QUERY   BIT(7)
+#define DUAL    BIT(8)
+#define SQUELCH BIT(7)
+#define MAX_CMD_LEN 128
+#define ACTIVE_STATION  1
+#define PASSIVE_STATION 0
+#define RoundFreq(a) ((int)((a)*1000.0+0.5)/1000.0)
+#define Freq2Idx(a) (int)(((a)-118.0) * 3040/(137.00-118.0)+0.5)
+#define BIT(n) (1 << (n))
+#define NAME_SIZE 30
+
+typedef union {
+  uint16_t intVal16;
+  uint8_t  intVal8[2];
+} IntConvertStruct;
+
+typedef struct {
+  double ActiveFrequency;    //active station frequency
+  double PassiveFrequency;   // passive (or standby) station frequency
+  TCHAR PassiveName[NAME_SIZE] ; // passive (or standby) station name
+  TCHAR ActiveName[NAME_SIZE] ;  //active station name
+  int Volume ;               // Radio Volume
+  int Squelch ;              // Radio Squelch
+  int Vox ;                  // Radio Intercom Volume
+  bool Changed;              // Parameter Changed Flag            (TRUE = parameter changed)
+  bool Enabled;              // Radio Installed d Flag            (TRUE = Radio found)
+  bool Dual;                 // Dual Channel mode active flag     (TRUE = on)
+  bool Enabled8_33;          // 8,33kHz Radio enabled             (TRUE = 8,33kHz)
+  bool RX;                   // Radio reception active            (TRUE = reception)
+  bool TX;                   // Radio transmission active         (TRUE = transmission)
+  bool RX_active;            // Radio reception on active station (TRUE = reception)
+  bool RX_standy;            // Radio reception on passive        (standby) station
+  bool lowBAT;               // Battery low flag                  (TRUE = Batt low)
+  bool TXtimeout;            // Timeout while transmission (2Min)
+} Radio_t;
+
+IntConvertStruct CRC;
+IntConvertStruct sFrequency;
+IntConvertStruct sStatus;
+
+volatile bool bSending = false;
+//End Einige Konstanten und Ausdruecke, das AR62xx hat ein Binaerprotokoll
+
+
+/**
+ * AR62xx device class.
+ *
+ * This class provides the interface to communicate with the AR62xx radio.
+ * The driver retransmits messages in case of a failure.
+ */
+class AR62xxDevice final : public AbstractDevice {
+  static constexpr unsigned CMD_TIMEOUT = 250; //!< Command timeout in millis.
+  static constexpr unsigned NR_RETRIES = 3; //!< Number of tries to send a command.
+
+  static constexpr char STX = 0x02; //!< Command start character.
+  static constexpr char ACK = 0x06; //!< Command acknowledged character.
+  static constexpr char NAK = 0x15; //!< Command not acknowledged character.
+  static constexpr char NO_RSP = 0; //!< No response received yet.
+
+  static constexpr size_t MAX_NAME_LENGTH = 10; //!< Max. radio station name length.
+  //! Port the radio is connected to.
+  Port &port;
+  //! Expected length of the message just receiving.
+  //size_t expected_msg_length{};
+  //! Buffer which receives the messages send from the radio.
+  StaticFifoBuffer<uint8_t, 256u> rx_buf;
+  //! Last response received from the radio.
+  uint8_t response;
```

Comment:
What does "response" mean? What are legal values, and what do they mean?

Reply:
response removed